sqlmesh/core/table_diff.py

```diff
@@ -75,6 +76,21 @@ class RowDiff(PydanticModel, frozen=True):
     model_name: t.Optional[str] = None
     decimals: int = 3
 
+    _types_resolved: t.ClassVar[bool] = False
+
+    def __new__(cls, *args: t.Any, **kwargs: t.Any) -> RowDiff:
+        if not cls._types_resolved:
+            cls._resolve_types()
+        return super().__new__(cls)
+
+    @classmethod
+    def _resolve_types(cls) -> None:
+        """Pandas is imported by type checking so we need to resolve the types with the real import before instantiating."""
+        import pandas as pd  # noqa
+
+        cls.model_rebuild()
+        cls._types_resolved = True
```

Comment:
This required some custom logic. The issue is that pandas is imported for type checking purposes but then when Pydantic instantiates the object it doesn't actually have pandas available and errors. So what I did was overrode new and had it rebuild the model on the first instantiation. Confirmed in tests that the issue would repro before and doesn't anymore after this change.